brave/app/scripts/controllers/bitgo.js

```diff
@@ -0,0 +1,197 @@
+const ObservableStore = require('obs-store')
+const bgUtil = require('brave-bitgo-client').util
+
+const supportedCoins = {
+  btc: 'Bitcoin',
+  bch: 'Bitcoin Cash',
+  btg: 'Bitcoin Gold',
+  bsv: 'Bitcoin SV',
+  zec: 'ZCash',
+  xlm: 'Stellar',
+  xrp: 'Ripple',
+  ltc: 'Litecoin',
+  dash: 'Dash',
+  algo: 'Algorand',
+  trx: 'Tron',
+  eos: 'Eos'
+}
+
+const uint8ToArrayBuf = (array) => {
+  return array.buffer.slice(array.byteOffset, array.byteLength + array.byteOffset)
+}
+
+const promiseGetSeed = (key) => {
+  return new Promise((resolve) => {
+    if (window.chrome && window.chrome.braveWallet) {
+      window.chrome.braveWallet.getBitGoSeed(key, resolve)
+    } else {
+      // polyfill
+      const seed = new Uint8Array(32)
+      window.crypto.getRandomValues(seed)
+      resolve(uint8ToArrayBuf(seed))
+    }
+  })
+}
+
+class BitGoController {
+  constructor (opts = {}) {
+    const initState = opts.initState || {}
+    this.proxyOrigin = opts.isProduction
+      ? 'https://bitgo-proxy.brave.com/'
+      : 'http://localhost:3000/'
+    this.store = new ObservableStore(initState)
+    this.keyringController = opts.keyringController
+    chrome.braveWallet.getProjectID((projectId) => {
+      this.braveServiceKey = projectId
+    })
+  }
+
+  request (path, opts) {
+    const url = `${this.proxyOrigin}${path}`
+    const headers = new window.Headers()
+    headers.append('x-brave-key', this.braveServiceKey)
+    const req = new window.Request(url, Object.assign({ headers }, opts))
+    return window.fetch(req)
+  }
+
+  async unlockAndSetKey (password) {
+    this.password = password
+    await this.keyringController.submitPassword(this.password)
+    this.encryptionKey = await this.keyringController._getSubkey('ethwallet-encryptor')
+  }
+
+  async createWallet (coin) {
+    if (!supportedCoins[coin]) {
+      throw new Error(`Coin type ${coin} is not supported.`)
+    }
+    const seed = await promiseGetSeed(uint8ToArrayBuf(this.encryptionKey))
+    const userKeychain = bgUtil.createKeychain(coin, seed)
+    const backupKeychain = bgUtil.createKeychain(coin, seed)
+    const req = {
+      coin,
+      userPub: userKeychain.pub,
+      backupPub: backupKeychain.pub
+    }
+    console.log({req})
```

Comment:
should this before this is ready for merge

Reply:
removed